src/test/java/rueppellii/backend2/tribes/building/service/BuildingServiceTest.java

```diff
@@ -0,0 +1,130 @@
+package rueppellii.backend2.tribes.building.service;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.context.jdbc.SqlGroup;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import rueppellii.backend2.tribes.building.exception.BuildingNotFoundException;
+import rueppellii.backend2.tribes.building.exception.UpgradeFailedException;
+import rueppellii.backend2.tribes.building.persistence.model.Building;
+import rueppellii.backend2.tribes.kingdom.persistence.model.Kingdom;
+import rueppellii.backend2.tribes.progression.persistence.ProgressionModel;
+
+import javax.transaction.Transactional;
+
+import static rueppellii.backend2.tribes.gameUtility.purchaseService.UpgradeConstants.BUILDING_MAX_LEVEL;
+
+@ExtendWith(SpringExtension.class)
+@ActiveProfiles("Test")
+@SpringBootTest
+@SqlGroup({@Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = "classpath:clear.sql"), @Sql(executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD, scripts = "classpath:data.sql"), @Sql(executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD, scripts = "classpath:clear.sql")})
+
+public class BuildingServiceTest {
+
+    @Autowired
+    private BuildingService buildingService;
+
+    private Kingdom testKingdom;
+    private Building testBuilding;
+    private Long buildingId;
+    private String buildingType;
+    private int buildingLevel;
+
+
```

Comment:
Try to follow this naming convention:
[the name of the tested feature]_[expected input / tested state]_[expected behavior]

like

registerNewUserAccount_ExistingEmailAddressGiven_ShouldThrowException()